openverse_catalog/dags/providers/provider_api_scripts/cleveland_museum.py

```diff
@@ -1,97 +1,57 @@
 import logging
+from typing import Dict, Optional
 
 from common.licenses import get_license_info
 from common.loader import provider_details as prov
-from common.requester import DelayedRequester
-from common.storage.image import ImageStore
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 
 
-LIMIT = 1000
-DELAY = 5.0
-RETRIES = 3
-PROVIDER = prov.CLEVELAND_DEFAULT_PROVIDER
-ENDPOINT = "http://openaccess-api.clevelandart.org/api/artworks/"
-
-delay_request = DelayedRequester(delay=DELAY)
-image_store = ImageStore(provider=PROVIDER)
-
-DEFAULT_QUERY_PARAMS = {"cc": "1", "has_image": "1", "limit": LIMIT, "skip": 0}
-
 logging.basicConfig(
     format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s", level=logging.INFO
 )
 logger = logging.getLogger(__name__)
 
 
-def main():
-    logger.info("Begin: Cleveland Museum API requests")
-    condition = True
-    offset = 0
-
-    while condition:
-        query_param = _build_query_param(offset)
-        response_json, total_images = _get_response(query_param)
-        if response_json is not None and total_images != 0:
-            batch = response_json["data"]
-            image_count = _handle_response(batch)
-            logger.info(f"Total images till now {image_count}")
-            offset += LIMIT
+class ClevelandDataIngester(ProviderDataIngester):
+    def __init__(
```

Comment:
Would the `date` parameter be passed to the `Ingester` class?

Reply:
Hm do you mean in general, the parameter that gets passed through `dated` DAGs like Metropolitan? For dated DAGs, you _can_ override the `__init__` to add the parameter. Good point, though -- I think it's worth adding `date` as an optional param, and providers that don't care about it can just ignore it 👍

Reply:
I was trying to rewrite one of the dated provider scripts using the ingester class to understand if there are problems that I'm not seeing simply from the code review, and this is where I stumbled, @stacimc :)